pydicom/dataset.py

```diff
@@ -2672,39 +2675,63 @@ def __eq__(self, other: object) -> bool:
             return True
 
         if isinstance(other, self.__class__):
+            # exclude all metadata from the comparison
+            # only the actual dataset content shall be compared
+            excludes = ("_dict", 'filename', "preamble", "file_meta",
+                        "is_implicit_VR", "is_little_endian", "timestamp")
             return (
```

Comment:
When I do `list(ds.__dict__.keys())` on a sample dataset, I see some more:
```python
['_parent_encoding',  'is_decompressed', 'read_little_endian', 'read_implicit_vr', 'read_encoding',  'parent', '_private_blocks', '_pixel_id', 'fileobj_type']
```
Should those not also be excluded?

Reply:
Well, after writing a test and seeing that it already passed I realized that we don't need an extra `FileDataset.__eq__` implementation, as `Dataset.__eq__` already makes sure that only tags are compared :astonished: